src/store/validators.ts

```diff
@@ -0,0 +1,43 @@
+import { FormEntry, ValidationFailure } from './nodeEditor';
+
+type FormInput = string | string[] | number;
```

Comment:
type cracking is lame, will probably rework this particular bit later, but its all encapsulated in this file